ui-speedspacechart/src/components/helpers/drawElements/speedLimitTags.ts

```diff
@@ -0,0 +1,183 @@
+  store: Store
+): Promise<tooltipInfos | null> => {
+  const {
+    speedLimitTags,
+    ratioX,
+    leftOffset,
+    layersDisplay: { electricalProfiles, powerRestrictions },
+    cursor,
+  } = store;
+
+  const { MARGIN_TOP, MARGIN_BOTTOM, MARGIN_LEFT, MARGIN_RIGHT } = MARGINS;
+
+  if (!speedLimitTags) return null;
+  clearCanvas(ctx, width, LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT);
+
+  ctx.save();
+  ctx.translate(leftOffset, 0);
+
+  const { maxPosition } = maxPositionValues(store);
+
+  const questionImage = await loadSvgImage(questionBlobUrl);
+  const alertFillImage = await loadSvgImage(alertFillBlobUrl);
+
+  let speedLimitTagsBackgroundColor = LINEAR_LAYERS_BACKGROUND_COLOR.FIRST;
+
+  if (electricalProfiles && powerRestrictions) {
+    speedLimitTagsBackgroundColor = LINEAR_LAYERS_BACKGROUND_COLOR.THIRD;
+  } else if (electricalProfiles || powerRestrictions) {
+    speedLimitTagsBackgroundColor = LINEAR_LAYERS_BACKGROUND_COLOR.SECOND;
+  }
+
+  drawSeparatorLinearLayer(ctx, 'rgba(0,0,0,0.1)', MARGINS, width, 1);
+  drawLinearLayerBackground(
+    ctx,
+    speedLimitTagsBackgroundColor,
+    MARGINS,
+    width,
+    MARGIN_BOTTOM,
+    LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT - 1
+  );
+
+  const RECT_HEIGHT = 17;
+  const yPosition = 12;
+  const RECTANGLE_SPACING = 1;
+  const TEXT_LEFT_PADDING = 4;
+  const TEXT_RIGHT_PADDING = 8;
+  const FIRST_TAG_LEFT_PADDING = 8;
+  const ICON_WIDTH = 16;
+  const ICON_HEIGHT = 16;
+  const ICON_OFFSET = 4;
+  const TEXT_PADDING_TOP = 1;
+
+  let tooltip: tooltipInfos | null = null;
+
+  speedLimitTags.boundaries.forEach((boundary, index) => {
+    const tag = speedLimitTags.values[index];
+    if (tag && tag.speed_limit_tags_type === 'tag') {
+      const { color, tag_name } = tag;
+      const x = positionOnGraphScale(boundary, maxPosition, width, ratioX, MARGINS);
+      const nextBoundary = speedLimitTags.boundaries[index + 1];
+
+      if (nextBoundary !== undefined) {
+        const tagWidth =
+          positionOnGraphScale(nextBoundary, maxPosition, width, ratioX, MARGINS) -
+          x -
+          RECTANGLE_SPACING;
+
+        const secondaryColor = colorDictionary[color] || color;
+
+        ctx.fillStyle = color;
+        ctx.strokeStyle = color;
+        ctx.lineWidth = 1;
+
+        ctx.beginPath();
+        ctx.rect(x, yPosition, tagWidth - RECTANGLE_SPACING, RECT_HEIGHT);
+        ctx.closePath();
+        ctx.fillRect(x, yPosition, tagWidth - RECTANGLE_SPACING, RECT_HEIGHT);
+
+        ctx.fillStyle = secondaryColor;
+
+        const textWidth =
+          ctx.measureText(tag_name).width +
+          TEXT_RIGHT_PADDING +
+          (index === 0 ? FIRST_TAG_LEFT_PADDING : 0);
+
+        ctx.fillRect(x + 1 + textWidth, yPosition, tagWidth - textWidth - 2, RECT_HEIGHT);
+        ctx.stroke();
+
+        if (tag_name === 'incompatible' || tag_name === 'missing_from_train') {
+          const image = tag_name === 'incompatible' ? alertFillImage : questionImage;
```

Comment:
Sketch : 
![Capture d’écran 2024-07-11 à 17 55 36](https://github.com/OpenRailAssociation/osrd-ui/assets/113104137/86d9ea0f-325c-4381-a844-a709ef5895ee)

PR :
![Capture d’écran 2024-07-11 à 17 58 21](https://github.com/OpenRailAssociation/osrd-ui/assets/113104137/d971752f-f6c4-458e-8c28-09be6058daef)

It seems like in sketch, the icon is filled with white. I'm not sure if it's the permanent style of the icon of if it's because we are hovering it. Either way, you should check that with @thibautsailly.

Reply:
not possible yet, we had a very hard time managing those SVG, we can only use the .svg files and can't modify it for now

Reply:
If the icon need to switch between filled and not filled icon (and if @thibautsailly wants it to change if hovered or not), do you think we could have an extra condition to test : "if cursor on the icon, image = 'filledIcon', if not image = 'normalIcon' "  ?